src/context.ts

```diff
@@ -13,9 +19,30 @@ type CreateContextParams = {
   connection?: unknown;
 };
 
+function getUserId(authorization: string): JwtVerificationResult {
+  if (!authorization) {
+    return { userId: null, error: 'Unhandled' };
+  }
+  const token = authorization.replace('Bearer ', '');
+  const verificationTokenResult = verify(token, JWT_SECRET, (err, decoded) => {
+    if (err) {
+      return { userId: null, error: err.name === 'TokenExpiredError' ? 'JwtExpired' : 'Unhandled' };
+    }
+    return decoded;
+  }) as unknown as JwtVerificationResult;
```

Comment:
Why do you have to retype it?

Reply:
because the native typing seems to be simply incorrect. It claims to return void, although the actual return is the decoded object.